src/Runner.Listener/JobDispatcher.cs

```diff
@@ -512,6 +512,7 @@ private async Task RunAsync(Pipelines.AgentJobRequestMessage message, string orc
                             var completedTask = await Task.WhenAny(renewJobRequest, workerProcessTask, Task.Delay(-1, jobRequestCancellationToken));
                             if (completedTask == workerProcessTask)
                             {
+                                Trace.Info(string.Join(Environment.NewLine, workerOutput));
```

Comment:
I'd propose to extract it to the variable`detailInfo = string.Join(Environment.NewLine, workerOutput);` and reuse here and in lines 531 and 534 :)

Reply:
SGTM! :) Amended, thanks!